daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1562,14 +1562,23 @@ private[lf] object SBuiltin {
         machine: Machine,
     ): Unit = {
       args.get(0) match {
-        case SAnyException(_, messageFunction, innerValue) =>
-          machine.ctrl = SEApp(messageFunction, Array(SEValue(innerValue)))
+        case SAnyException(ty, innerValue) =>
+          machine.ctrl = SEApp(exceptionMessage(ty), Array(SEValue(innerValue)))
```

Comment:
I don’t like computing this at runtime all that much. I’m not really worried about performance and it seems nicer than what we had before but if someone has a better proposal I’m all ears.

Reply:
This function should be called really infrequently, so I think it should be fine.

Reply:
I agree. Can't imagine there is any perf issue here.